### PR TITLE
i think ? is bettern than string in understanding

### PR DESCRIPTION
**Pull request by RainPoetry**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by RainPoetry** [comment]
i think ? is bettern than string in understanding

**Comment by RainPoetry** [comment]
instead String of ?

File: clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java [hunk 1]
```diff
@@ -342,7 +342,7 @@ public void logUnused() {
      * @param configOverrides Configuration overrides to use.
      * @return The list of configured instances
      */
-    public <T> List<T> getConfiguredInstances(List<String> classNames, Class<T> t, Map<String, Object> configOverrides) {
+    public <T> List<T> getConfiguredInstances(List<?> classNames, Class<T> t, Map<String, Object> configOverrides) {
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java:345]: We want the class names to be a string, so this change doesn't seem to be desirable.

**Review by ijuma — COMMENTED** [review]
Thanks for your contribution. Not sure we want to make this change.

**Comment by mimaison** [comment]
`AbstractConfig` is a public API. I don't think we want to do this change. Closing this old PR.

**Pull request closed without merging**